packages/teams-js/test/public/settings.spec.ts

```diff
@@ -71,46 +57,131 @@ describe('settings', () => {
+          await utils.initializeWithContext(context);
+          settings.getSettings(settings => {
+            expect(settings).toBe(expectedSettings);
+          });
+
+          const message = utils.findMessageByFunc('settings.getSettings');
+          expect(message).not.toBeNull();
+
+          utils.respondToMessage(message, expectedSettings);
+        });
+      }
     });
   });
 
-  it('should successfully set settings', () => {
-    utils.initializeWithContext('settings').then(() => {
-      const settingsObj: settings.Settings = {
-        suggestedDisplayName: 'someSuggestedDisplayName',
-        contentUrl: 'someContentUrl',
-        websiteUrl: 'someWebsiteUrl',
-        entityId: 'someEntityId',
-      };
-      settings.setSettings(settingsObj);
+  describe('Testing settings.setSettings function', () => {
+    const allowedContexts = [FrameContexts.content, FrameContexts.settings, FrameContexts.sidePanel];
+    const settingsObj: settings.Settings = {
+      suggestedDisplayName: 'someSuggestedDisplayName',
+      contentUrl: 'someContentUrl',
+      websiteUrl: 'someWebsiteUrl',
+      entityId: 'someEntityId',
+    };
+
+    it('settings.setSettings should not allow calls before initialization', () => {
+      expect(() => {
+        settings.setSettings(settingsObj);
+      }).toThrowError('The library has not yet been initialized');
+    });
 
-      const message = utils.findMessageByFunc('settings.setSettings');
-      expect(message).not.toBeNull();
-      expect(message.args.length).toBe(1);
-      expect(message.args[0]).toBe(settingsObj);
+    Object.values(FrameContexts).forEach(context => {
+      if (!allowedContexts.some(allowedContexts => allowedContexts == context)) {
+        it(`settings.setSettings does not allow calls from ${context} context`, async () => {
+          await utils.initializeWithContext(context);
+          expect(() => {
+            settings.setSettings(settingsObj);
+          }).toThrowError(
+            `This call is only allowed in following contexts: ${JSON.stringify(
+              allowedContexts,
+            )}. Current context: "${context}".`,
+          );
+        });
+      } else {
+        it(`settings.setSettings should successfully set settings from ${context} context`, async () => {
+          await utils.initializeWithContext(context);
+          settings.setSettings(settingsObj);
+          const message = utils.findMessageByFunc('settings.setSettings');
+          expect(message).not.toBeNull();
+          expect(message.args.length).toBe(1);
+          expect(message.args[0]).toBe(settingsObj);
+        });
+      }
     });
   });
 
-  it('should successfully register a save handler', () => {
-    utils.initializeWithContext('settings').then(() => {
+  describe('Testing settings.registerOnSaveHandler function', () => {
+    const allowedContexts = [FrameContexts.settings];
+    it('settings.registerOnSaveHandler should not allow calls before initialization', () => {
```

Comment:
nit: newline before the start of the test case

Reply:
Added in the latest commit.